.github/workflows/main.yml

```diff
@@ -15,47 +15,75 @@ on:
 jobs:
   build:
     strategy:
-      fail-fast: false
+      fail-fast: true
       matrix:
         os: [ubuntu-latest]
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         include:
           - os: windows-latest
-            python-version: "3.11"
+            python-version: "3.12"
           - os: macos-latest
-            python-version: "3.11"
+            python-version: "3.12"
     runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Clone the repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install ".[test]"
 
+      - name: Test with pytest
+        run: |
+          pytest tests
+
+  linting:
+    runs-on: ubuntu-latest
```

Comment:
I agree with @qubixes. We want to offer iBridges for all three operating systems an in the past the linting has pointed us to some incompatibilities. So I would keep it in. When/if we extend the coverage of the integration tests we can reduce the linting to linux.

Reply:
I moved the pylint and mypy up to the orginial place again. 

In theory you are right it should be enough to test for lunix should be enough. However,  in the past development cycles we encountered some descrepancies between the linting/mypy results, most probably due to that we do not fully understand the whole stack from OS, through PRC to the iRODS server. Nevertheless, the linting in this section of the workflow prevented us from releasing buggy software for MAc and Windows. So I would like to keep them here.